src/var.ml

```diff
@@ -0,0 +1,187 @@
+
+    ; "path-no-dep", deleted_in ~version:(1, 0) Path_no_dep
+    ]
+    |> String.Map.of_list_exn
+
+  let create_vars ~(context : Context.t) ~cxx_flags =
+    let ocamlopt =
+      match context.ocamlopt with
+      | None -> Path.relative context.ocaml_bin "ocamlopt"
+      | Some p -> p
+    in
+    let string s = values [Value.String s] in
+    let path p = values [Value.Path p] in
+    let make =
+      match Bin.make with
+      | None   -> string "make"
+      | Some p -> path p
+    in
+    let cflags = context.ocamlc_cflags in
+    let strings s = values (Value.L.strings s) in
+    let lowercased =
+      [ "cpp"            , strings (context.c_compiler :: cflags @ ["-E"])
+      ; "cc"             , strings (context.c_compiler :: cflags)
+      ; "cxx"            , strings (context.c_compiler :: cxx_flags)
+      ; "ocaml"          , path context.ocaml
+      ; "ocamlc"         , path context.ocamlc
+      ; "ocamlopt"       , path ocamlopt
+      ; "arch_sixtyfour" , string (string_of_bool context.arch_sixtyfour)
+      ; "make"           , make
+      ; "root"           , values [Value.Dir context.build_dir]
+      ] in
+    let uppercased =
+      List.map lowercased ~f:(fun (k, _) ->
+        (String.uppercase k, renamed_in ~new_name:k ~version:(1, 0))) in
+    let vars =
+      [ "-verbose"       , values []
+      ; "pa_cpp"         , strings (context.c_compiler :: cflags
+                                    @ ["-undef"; "-traditional";
+                                       "-x"; "c"; "-E"])
+      ; "ocaml_bin"      , path context.ocaml_bin
+      ; "ocaml_version"  , string context.version_string
+      ; "ocaml_where"    , string (Path.to_string context.stdlib_dir)
+      ; "null"           , string (Path.to_string Config.dev_null)
+      ; "ext_obj"        , string context.ext_obj
+      ; "ext_asm"        , string context.ext_asm
+      ; "ext_lib"        , string context.ext_lib
+      ; "ext_dll"        , string context.ext_dll
+      ; "ext_exe"        , string context.ext_exe
+      ; "profile"        , string context.profile
+      ]
+    in
+    let ocaml_config =
+      List.map (Ocaml_config.to_list context.ocaml_config) ~f:(fun (k, v) ->
+        ("ocaml-config:" ^ k,
+         match (v : Ocaml_config.Value.t) with
+         | Bool   x -> string (string_of_bool x)
+         | Int    x -> string (string_of_int x)
+         | String x -> string x
+         | Words  x -> strings x
+         | Prog_and_args x -> strings (x.prog :: x.args)))
+    in
+    [ ocaml_config
+    ; static_vars
+    ; lowercased
+    ; uppercased
+    ; vars
+    ]
+    |> List.concat
+    |> String.Map.of_list_exn
+
+  let static_vars = String.Map.of_list_exn static_vars
+
+  let rec expand t ~syntax_version ~var =
+    let name = String_with_vars.Var.name var in
+    Option.bind (String.Map.find t name) ~f:(fun v ->
+      let what var =
+        sprintf "Variable %s" (String_with_vars.Var.to_string var) in
```

Comment:
We should check here if `var` is a form or variable In order to display a different error message.  If it's a form we should display: `The %{<var-name>:...} form`, i.e. with the payload replaced by `...` so  that it's clear we are talking about the form in general and not just this particular instance.

Reply:
Ah yes. Makes sense. I only made this fix for a particular error message.